src/main/java/com/educare/unitylend/dao/WalletRepository.java

```diff
@@ -19,26 +14,102 @@
 @Repository
 public interface WalletRepository {
 
-    @Select("select * from wallet")
+    static final String SELECT_WALLETS = "select * from wallet";
+
+    static String getWalletInfoQuery(String userId) {
+        StringBuilder sqlQueryBuilder = new StringBuilder();
+
+        sqlQueryBuilder.append("SELECT w.walletid, w.userid, w.balance FROM wallet w JOIN tempuser u ON w.userid = u.userid WHERE 1=1");
+        if (userId != null) {
+            sqlQueryBuilder.append(" AND w.userid = '").append(userId).append("'");
+        }
+
+        return sqlQueryBuilder.toString();
+    }
+
+
+    public static String generateWalletInTableQuery(String userId) {
+        StringBuilder sqlQueryBuilder = new StringBuilder();
+
+        sqlQueryBuilder.append("INSERT INTO wallet (walletid, balance, userid) VALUES (uuid_generate_v4(), 0, ");
+        if (userId != null) {
+            sqlQueryBuilder.append("'").append(userId).append("'");
+        } else {
+            sqlQueryBuilder.append("NULL");
+        }
+        sqlQueryBuilder.append(")");
+
+        return sqlQueryBuilder.toString();
+    }
+
+    public static String getUserIdWithWalletQuery(@Param("ex") Wallet ex) {
+        StringBuilder sqlQueryBuilder = new StringBuilder();
+
+        sqlQueryBuilder.append("SELECT u.* FROM tempuser u WHERE u.userid = (SELECT userid FROM wallet WHERE walletid = ");
+        if (ex != null && ex.getWalletId() != null) {
+            sqlQueryBuilder.append("'").append(ex.getWalletId()).append("'");
+        } else {
+            sqlQueryBuilder.append("NULL");
+        }
+        sqlQueryBuilder.append(")");
+
+        return sqlQueryBuilder.toString();
+    }
+
+    public static String getWalletByIdQuery(@Param("walletid") String walletid) {
+        StringBuilder sqlQueryBuilder = new StringBuilder();
+
+        sqlQueryBuilder.append("SELECT * FROM wallet WHERE walletid = ");
+        if (walletid != null) {
+            sqlQueryBuilder.append("'").append(walletid).append("'");
+        } else {
+            sqlQueryBuilder.append("NULL");
+        }
+
```

Comment:
The below code is not making sense, 
Why are we comparing with null;

Instead throw exception when walletId is null 
```
if (walletid == null) {
        throw new IllegalArgumentException("walletid cannot be null");
    }
```

Reply:
Add similar exception handling in all the other methods where params are necessary